modules/jpg/image_loader_libjpeg_turbo.cpp

```diff
@@ -0,0 +1,191 @@
+		// Proceed anyway and convert to rgb8?
+		tj3Destroy(tj_instance);
+		return ERR_UNAVAILABLE;
+	}
+
+	TJPF tj_pixel_format;
+	Image::Format gd_pixel_format;
+	if (colorspace == TJCS_GRAY) {
+		tj_pixel_format = TJPF_GRAY;
+		gd_pixel_format = Image::FORMAT_L8;
+	} else {
+		// Force everything else (RGB, CMYK etc) into RGB8.
+		tj_pixel_format = TJPF_RGB;
+		gd_pixel_format = Image::FORMAT_RGB8;
+	}
+
+	Vector<uint8_t> data;
+	data.resize(width * height * tjPixelSize[tj_pixel_format]);
+
+	if (tj3Decompress8(tj_instance, p_buffer, p_buffer_len, data.ptrw(), 0, tj_pixel_format) < 0) {
+		tj3Destroy(tj_instance);
+		return ERR_FILE_CORRUPT;
+	}
+
+	tj3Destroy(tj_instance);
+	p_image->set_data(width, height, false, gd_pixel_format, data);
+	return OK;
+}
+
+Error ImageLoaderLibJPEGTurbo::load_image(Ref<Image> p_image, Ref<FileAccess> f, BitField<ImageFormatLoader::LoaderFlags> p_flags, float p_scale) {
+	Vector<uint8_t> src_image;
+	uint64_t src_image_len = f->get_length();
+	ERR_FAIL_COND_V(src_image_len == 0, ERR_FILE_CORRUPT);
+	src_image.resize(src_image_len);
+
+	uint8_t *w = src_image.ptrw();
+
+	f->get_buffer(&w[0], src_image_len);
+
+	Error err = jpeg_turbo_load_image_from_buffer(p_image.ptr(), w, src_image_len);
+
+	return err;
+}
+
+void ImageLoaderLibJPEGTurbo::get_recognized_extensions(List<String> *p_extensions) const {
+	p_extensions->push_back("jpg");
+	p_extensions->push_back("jpeg");
+}
+
+static Ref<Image> _jpeg_turbo_mem_loader_func(const uint8_t *p_png, int p_size) {
+	Ref<Image> img;
+	img.instantiate();
+	Error err = jpeg_turbo_load_image_from_buffer(img.ptr(), p_png, p_size);
+	ERR_FAIL_COND_V(err, Ref<Image>());
+	return img;
+}
+
+static Vector<uint8_t> _jpeg_turbo_buffer_save_func(const Ref<Image> &p_img, float p_quality) {
+	Vector<uint8_t> output;
+
+	ERR_FAIL_COND_V(p_img.is_null() || p_img->is_empty(), output);
+
+	Ref<Image> image = p_img->duplicate();
+	if (image->is_compressed()) {
+		Error error = image->decompress();
+		ERR_FAIL_COND_V_MSG(error != OK, output, "Couldn't decompress image.");
+	}
+
+	if (image->get_format() != Image::FORMAT_RGB8) {
+		// Allow grayscale L8?
+		image = image->duplicate();
+		image->convert(Image::FORMAT_RGB8);
+	}
+
+	tjhandle tj_instance = tj3Init(TJINIT_COMPRESS);
+	ERR_FAIL_COND_V_MSG(tj_instance == NULL, output, "Couldn't create tjhandle");
+
+	if (tj3Set(tj_instance, TJPARAM_QUALITY, (int)(p_quality * 100)) < 0) {
+		tj3Destroy(tj_instance);
+		ERR_FAIL_V_MSG(output, "Couldn't set jpg quality");
+	}
+
+	if (tj3Set(tj_instance, TJPARAM_PRECISION, 8) < 0) {
+		tj3Destroy(tj_instance);
+		ERR_FAIL_V_MSG(output, "Couldn't set jpg precision");
+	}
+
+	if (tj3Set(tj_instance, TJPARAM_SUBSAMP, TJSAMP_420) < 0) {
+		tj3Destroy(tj_instance);
+		ERR_FAIL_V_MSG(output, "Couldn't set jpg subsamples");
+	}
+
+	// If the godot image format is `Image::FORMAT_L8` we could set the appropriate
+	// colour space here rather than defaulting to RGB.
```

Comment:
I'm surprised `codespell` doesn't complain about this, as it's meant to standardize on en_US spelling.
```suggestion
	// color space here rather than defaulting to RGB.
```